backend/layers/processing/schema_migration.py

```diff
@@ -188,15 +188,17 @@ def publish_and_cleanup(self, collection_version_id: str, can_publish: bool) ->
 
         # Get the datasets that were processed
         extra_info = self._retrieve_sfn_response("publish_and_cleanup", collection_version.collection_id.id)
-        processed_datasets = [d["dataset_version_id"] for d in extra_info["datasets"]]
+        processed_datasets = {d["dataset_id"]: d["dataset_version_id"] for d in extra_info["datasets"]}
 
         # Process datasets errors
         for dataset in collection_version.datasets:
+            dataset_id = dataset.dataset_id.id
             dataset_version_id = dataset.version_id.id
-            if dataset_version_id not in processed_datasets:
+            if dataset_id not in processed_datasets:
                 # skip datasets that were not processed
                 continue
-            key_prefix = self.get_key_prefix(dataset_version_id)
+            # filepath to clean-up uses dataset_version_id from the replaced version; accessing with dataset_id as key
+            key_prefix = self.get_key_prefix(processed_datasets[dataset_id])
```

Comment:
Something is still wrong about this. This `dataset_version_id` is for the unmigrated dataset. We do want to  delete the /migrated.h5ad from it.

Reply:
Isn't that where the migrated.h5ad is uploaded? The unmigrated dataset's directory? At the time we run migrate, we don't know the new dataset's version ID yet.

Reply:
You're right, the syntax is confusing. Can we break out `processed_datasets[dataset_id]` in a variable name `old_dataset_version_id`